iceoryx_posh/test/moduletests/test_popo_server_port_user.cpp

```diff
@@ -0,0 +1,732 @@
+}
+
+TEST_F(ServerPort_test, HasClientsWithClientIsTrue)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "3e0e81dd-8d39-435f-b7a7-8f7121356fa4");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    addClientQueue(sut);
+
+    EXPECT_TRUE(sut.portUser.hasClients());
+}
+
+TEST_F(ServerPort_test, HasClientsWithNoClientsButIntermediatelyHavingClientsIsFalse)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "2f890938-e6ae-4dce-a4e8-acbed1934153");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    addClientQueue(sut);
+    removeClientQueue(sut);
+
+    EXPECT_FALSE(sut.portUser.hasClients());
+}
+
+// END hasClients tests
+
+// BEGIN hasNewRequests tests
+
+TEST_F(ServerPort_test, HasNewRequestsWithoutOfferIsFalse)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "acac09bb-4e80-426a-b23b-16e55c1ce1c8");
+    auto& sut = serverPortWithoutOfferOnCreate;
+    EXPECT_FALSE(sut.portUser.hasNewRequests());
+}
+
+TEST_F(ServerPort_test, HasNewRequestsWithNoRequestsIsFalse)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "e7aa181d-bd4f-46f5-ad57-2cc54a4ec93f");
+    auto& sut = serverPortWithOfferOnCreate;
+    EXPECT_FALSE(sut.portUser.hasNewRequests());
+}
+
+TEST_F(ServerPort_test, HasNewRequestsWithOneRequestIsTrue)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "0235ce49-96b3-41c0-aac1-e0d6f6bc2b1f");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    constexpr uint64_t NUMBER_OF_REQUESTS{1U};
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+
+    EXPECT_TRUE(sut.portUser.hasNewRequests());
+}
+
+TEST_F(ServerPort_test, HasNewRequestsWithNoRequestsButPreviouslyHavingOneIsFalse)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "29b0a347-2439-488d-ba13-962c7acffac1");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    constexpr uint64_t NUMBER_OF_REQUESTS{1U};
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+    IOX_DISCARD_RESULT(sut.portUser.getRequest());
+
+    EXPECT_FALSE(sut.portUser.hasNewRequests());
+}
+
+TEST_F(ServerPort_test, HasNewRequestsWithOneRequestButIntermediatelyHavingNoneIsTrue)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "a9fc0d89-7b97-48c4-8ba4-e4b22355116b");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    constexpr uint64_t NUMBER_OF_REQUESTS{1U};
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+    IOX_DISCARD_RESULT(sut.portUser.getRequest());
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+
+    EXPECT_TRUE(sut.portUser.hasNewRequests());
+}
+
+TEST_F(ServerPort_test, HasNewRequestsWithMultipleRequestsIsTrue)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "30e36e69-4f23-41ee-aac7-d34240f075ae");
+    auto& sut = serverPortWithOfferOnCreate;
+
+    constexpr uint64_t NUMBER_OF_REQUESTS{2U};
+    pushRequests(sut.requestQueuePusher, NUMBER_OF_REQUESTS);
+
+    EXPECT_TRUE(sut.portUser.hasNewRequests());
+}
+
+TEST_F(ServerPort_test, HasNewRequestsWithMultipleRequestsAndAllButOneRemovedIsTrue)
```

Comment:
Could you add a test for the limits. `HasNewRequestsIsTrueWithMaximumAmountOfRequests`.

Reply:
I added a test. I'm not quite sure how much these functions need to be tested since they are delegating a call to the `ChunkReceiver` which is already extensively tested.

Reply:
You are most likely right but better safe than sorry.